_posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc

```diff
@@ -0,0 +1,248 @@
+  Description: Starts a standalone instance of WildFly and deploys the
+    application to the server. The deployment type must be a WAR. Once the
+    server is running, the source directories are monitored for changes. If
+    required the sources will be compiled and the deployment may be redeployed.
+    Note that changes to the POM file are not monitored. If changes are made
+    the POM file, the process will need to be terminated and restarted. Note
+    that if a WildFly Bootable JAR is packaged, it is ignored by this goal.
+  Implementation: org.wildfly.plugin.dev.DevMojo
+  Language: java
+  Bound to phase: package
+
+  Available parameters:
+
+    ...
+
+    debug (Default: false)
+      User property: wildfly.debug
+      Starts the server with debugging enabled.
+
+    debugHost (Default: *)
+      User property: wildfly.debug.host
+      Sets the hostname to listen on for debugging. An * means all hosts.
+
+    debugPort (Default: 8787)
+      User property: wildfly.debug.port
+      Sets the port the debugger should listen on.
+
+    debugSuspend (Default: false)
+      User property: wildfly.debug.suspend
+      Indicates whether the server should suspend itself until a debugger is
+      attached.
+
+---
+
+wildfly:run
+  Description: Starts a standalone instance of WildFly and deploys the
+    application to the server. This goal will block until cancelled or a
+    shutdown is invoked from a management client. Note that if a WildFly
+    Bootable JAR is packaged, it is ignored by this goal.
+  Implementation: org.wildfly.plugin.server.RunMojo
+  Language: java
+  Before this goal executes, it will call:
+    Phase: 'package'
+
+  Available parameters:
+
+    ...
+
+    debug (Default: false)
+      User property: wildfly.debug
+      Starts the server with debugging enabled.
+
+    debugHost (Default: *)
+      User property: wildfly.debug.host
+      Sets the hostname to listen on for debugging. An * means all hosts.
+
+    debugPort (Default: 8787)
+      User property: wildfly.debug.port
+      Sets the port the debugger should listen on.
+
+    debugSuspend (Default: false)
+      User property: wildfly.debug.suspend
+      Indicates whether the server should suspend itself until a debugger is
+      attached.
+----
+
+As the document written in above, both the `wildfly:dev` and the `wildfly:run` goals support the `debug` mode, and the property to activate it is the `wildfly.debug` option.
+
+[source,bash]
+----
+$ mvn wildfly:run -Dwildfly.debug=true
+----
+
+And from the server output I can see the debug options are added:
+
+....
+[INFO] JAVA_OPTS : -Xms64m -Xmx512m -Djava.net.preferIPv4Stack=true -Djava.awt.headless=true -Djboss.modules.system.pkgs=org.jboss.byteman -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:8787 --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-modules=java.se
+....
```

Comment:
This needs to be ---- and not ....

With the .... this renders as one very long line, and not in a yellow block with a horizontal scroll bar on the bottom. The one line is twice as wide as the rest of the page.

Reply:
done